python/python/bystro/ancestry/listener.py

```diff
@@ -0,0 +1,105 @@
+"""Provide a worker for the ancestry model."""
+import argparse
+import logging
+from pathlib import Path
+
+from ruamel.yaml import YAML
+
+from bystro.ancestry.ancestry_types import AncestryResponse, AncestrySubmission
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.beanstalkd.worker import ProgressPublisher, QueueConf, get_progress_reporter, listen
+
+logger = logging.getLogger()
+
+ANCESTRY_TUBE = "ancestry"
+
+
+class AncestryJobData(BaseMessage, frozen=True):
+    """Wrap an AncestrySubmission in a BaseMessage for beanstalk."""
+
+    ancestry_submission: AncestrySubmission
+
+
+class AncestryJobCompleteMessage(CompletedJobMessage, frozen=True, kw_only=True):
+    """Wrap an AncestryResponse in a CompletedJobMessage for beanstalk."""
+
+    results: AncestryResponse
+
+
+def _load_queue_conf(queue_conf_path: str) -> QueueConf:
+    with Path(queue_conf_path).open(encoding="utf-8") as queue_config_file:
+        raw_queue_conf = YAML(typ="safe").load(queue_config_file)
+    beanstalk_conf = raw_queue_conf["beanstalkd"]
+    return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
+
+
+def _infer_ancestry(
```

Comment:
I would probably do the business logic in Ray, so that we can scale out to N machines

Reply:
ok, IIUC that sounds like a task for the next PR unless there's anything currently actionable?